implementations/scripts/SourceCode.py

```diff
@@ -0,0 +1,101 @@
+class IsSourceCode:
+
+    def __init__(self, source_code_exclude_list=None, method="naive"):
+        """
+        A IsSourceCode object is used to help define source code.
+        :param source_code_exclude_list: a list of either
+            file extensions or directories to exclude
+        :param method: a string which can be "naive",
+            "folder_exclude" or "extension_exclude"
+        For example,
+        source_code_exclude_list = ["py", "other", "gitignore"]
+        or source_code_exclude_list = ["tests/", "bin/", "drivers/base/"]
+        """
+        self.source_code_exclude_list = source_code_exclude_list
+        self.method = method
+
+    def is_source_code(self, commit):
+        """
+        Given a commit structure, which is a dictionary returned
+        by the _summary function, and given a list of files to exclude
+        using source_code_exclude_list while instantiating an object,
+        decide whether all the files in a commit are to be excluded or not.
+
+        :param commit: a commit structure, returned by the _summary method.
+        """
+        options_dict = {
+            "naive": self._naive_implementation,
```

Comment:
I think each of these options would be the semantics of one of those classes that I mention above.